pure tests fail

running `tox -epy37-pure` leads to the following exception:

```
Failure in test test_compilation (Persistence.tests.test_persistent.PersistenceTest)
Traceback (most recent call last):
  File ".../3.7/lib/python3.7/unittest/case.py", line 59, in testPartExecutor
    yield
  File ".../3.7/lib/python3.7/unittest/case.py", line 615, in run
    testMethod()
  File "/Users/mac/vcs/gocept/Persistence/src/Persistence/tests/test_persistent.py", line 235, in test_compilation
    from Persistence import _Persistence
  File ".../3.7/lib/python3.7/unittest/case.py", line 203, in __exit__
    self._raiseFailure("{} not raised".format(exc_name))
  File ".../3.7/lib/python3.7/unittest/case.py", line 135, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: (<class 'AttributeError'>, <class 'ImportError'>) not raised
```

I get the same exception for all the other supported CPython versions.
They do not show up on TravisCI/Appveyor because there we only test the C version.

Can anyone reproduce this failure?

--- a/.coveragerc
+++ b/.coveragerc
@@ -1,7 +1,13 @@
 [run]
 branch = True
-source = src
+source = Persistence
 
 [report]
 precision = 2
 show_missing = True
+
+[paths]
+source =
+  src/
+  .tox/*/lib/*/site-packages/
+  .tox/pypy*/site-packages/

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,16 +1,21 @@
 language: python
+dist: xenial
 python:
     - 2.7
     - 3.5
     - 3.6
+    - 3.7
     - pypy
     - pypy3
 matrix:
     include:
-        - python: "3.7"
-          dist: xenial
+        - name: "2.7-pure"
+          python: "2.7"
+          env: PURE_PYTHON=1
+        - name: "3.7-pure"
+          python: "3.7"
+          env: PURE_PYTHON=1
 install:
-    - pip install six==1.10.0 # force to avoid conflict with zc.recipe.testrunner
     - pip install zc.buildout
     - pip install coveralls coverage
     - buildout bootstrap

--- a/tox.ini
+++ b/tox.ini
@@ -13,15 +13,14 @@ envlist =
     coverage-report,
 
 [testenv]
+install_command = pip install --no-cache-dir --no-binary=:all: {opts} {packages}
 commands =
-    {envbindir}/buildout -c {toxinidir}/buildout.cfg buildout:directory={envdir} buildout:develop={toxinidir} bootstrap
-    {envbindir}/buildout -c {toxinidir}/buildout.cfg buildout:directory={envdir} buildout:develop={toxinidir}
-    coverage run {envbindir}/test {posargs:-vc1}
-skip_install = true
+    coverage run -m zope.testrunner --test-path={envsitepackagesdir} -s Persistence {posargs:-vc1}
 deps =
-    zc.buildout
+    zope.testrunner
     coverage
 setenv =
+    PIP_NO_CACHE = 1
     COVERAGE_FILE=.coverage.{envname}
 
 [testenv:py27-pure]
@@ -29,24 +28,28 @@ basepython =
     python2.7
 setenv =
     PURE_PYTHON = 1
+    COVERAGE_FILE=.coverage.{envname}
 
 [testenv:py35-pure]
 basepython =
     python3.5
 setenv =
     PURE_PYTHON = 1
+    COVERAGE_FILE=.coverage.{envname}
 
 [testenv:py36-pure]
 basepython =
     python3.6
 setenv =
     PURE_PYTHON = 1
+    COVERAGE_FILE=.coverage.{envname}
 
 [testenv:py37-pure]
 basepython =
     python3.7
 setenv =
     PURE_PYTHON = 1
+    COVERAGE_FILE=.coverage.{envname}
 
 [testenv:coverage-report]
 basepython = python3.6

